internal/contourconfiguration/contourconfiguration.go

```diff
@@ -0,0 +1,77 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package contourconfiguration
```

Comment:
do you see this package eventually ~replacing `pkg/config`?

Re: naming, I'd be fine shortening this to just "config", though I recognize that's a pretty generic name. OK with whatever everyone prefers.

Reply:
mostly just wanted a place to put helpers that we may need for using the ContourConfiguration CRD types, similar to what we have for `internal/gatewayapi` or `internal/ingressclass`

didnt put it in the `apis/v1alpha1` package since I wasn't sure if we wanted to keep tests there and since it uses `internal` types

Reply:
When I was reading through some code, `contourconfiguration` (this module) and 
`contourConfiguration` (the way we refer to the actual ContourConfiguration object) feel very similar and easy to get wrong. What do you think about calling this `contourconfig` instead? It's a little shorter and more distinct.

I would definitely class this as a nit though, non-blocking.